engine.py

```diff
@@ -380,6 +389,9 @@ def __init__(self, start_sq, end_sq, board):
         self.end_col = end_sq[1] # intended column destiantion of piece to e moved
         self.piece_moved = board[self.start_row][self.start_col] # actual piece moved
         self.piece_captured = board[self.end_row][self.end_col] # opponent piece if any on the destination square
+        self.isPawnPromotion = False
```

Comment:
make the variable name snake_case instead of camelCase for consistency for the rest of the variables in this project